### PR TITLE
Add Jenkins pipeline support in ircbot-plugin - an ircNotify() step

### PR DESCRIPTION
**Pull request by jimklimov**
Status: "Works for us, review and merge welcome" ;)

I wanted to post this separately from the style cleanup done in #20, but it was too complicated to superficially make incompatible patch sets just for the sake of keeping PRs separate. They are in different commits and I hope this suffices to keep the review chores relatively simple.

Until the #20 is merged, reviewers are welcome to compare the PR'ed branch contents added on top of that cleanup via https://github.com/jimklimov/ircbot-plugin/compare/fix-style...jimklimov:pipeline-support

Feature-wise, this builds upon and requires https://github.com/jenkinsci/instant-messaging-plugin/pull/16 which implements the pipeline support (note semi-cosmetic fixes proposed in https://github.com/akomakom/instant-messaging-plugin/pull/1 not merged so far), and replicates the consumer of that support originally done for the Jabber plugin in https://github.com/jenkinsci/jabber-plugin/pull/18 with a slight adaptation for the ircbot plugin.

Note that architecturally here the pipeline Step interface is provided by the protocol plugins (irc, jabber) to send notifications over this or that medium, but the real bulk of implementation is in the commit instant-messaging plugin.

When it hopefully gets to merging this PR ecosystem, please take care to bump the published releases and fix the version requirements in the `pom.xml` files, so sufficiently advanced dependencies get used by maven and the Jenkins plugin installer.

Unfortunately, not being much of a Java developer, I did not find any better way to build the wad of plugins than to `mvn clean package install` the codebase which includes https://github.com/jenkinsci/instant-messaging-plugin/pull/16 and a modified `pom.xml` with `<version>1.36-SNAPSHOT-PIPELINES</version>` so I had the specially tagged JAR in my local maven repository, and then I could build, test and use the ircbot codebase which requires this tagged version. This would probably preclude automated Jenkins builds of the PR, however (and would be an additional reason to need the post-merge changes outlined above).

### CODE REVIEW

**Comment by jimklimov** [comment]
CC @casz @kutzi @akomakom

**Comment by casz** [comment]
Not sure why I am cced :confused:

**Comment by kutzi** [comment]
@jimklimov I'm still listed as maintainer for this plugin, but I'm not really acting anymore - sorry.
As I'm not part of the jenkins-dev mailing list anymore, I cannot even ask to remove me as maintainer from the wiki page ...

**Comment by kutzi** [comment]
That being said: I think you'll have the best chances if you ask on the jenkins-dev mailing list for support.
Maybe someone else wants to step up as the maintainer? Maybe you want to?

**Comment by jimklimov** [comment]
@casz : sorry, a miss on my side

You were very helpful in other reviews on other components, and I must have mixed up which developments you were part of. Still, every day for us something new, to dig in, right? ;)

**Comment by jimklimov** [comment]
Alas, the CI tests would be red until the https://github.com/jenkinsci/instant-messaging-plugin/pull/16 (and follow-up, see above) is merged.

File: src/main/java/hudson/plugins/ircbot/IrcNotifyStep.java [hunk 1]
```diff
@@ -56,6 +59,10 @@
     private BuildToChatNotifier buildToChatNotifier = new DefaultBuildToChatNotifier();
     private MatrixJobMultiplier matrixNotifier = MatrixJobMultiplier.ONLY_PARENT;
 
+    // Instead of build status messages, send an arbitrary message to specified
```
Comment (jimklimov) [inline src/main/java/hudson/plugins/ircbot/IrcNotifyStep.java]: Note: As subsequent co-evolution of ircbot and instant-messaging plugins, maybe makes sense to evict this feature there, similarly to `extraMessage` done in @akomakom's branch.

**Comment by jimklimov** [comment]
At this time, codebase of this PR works against the codebase in release 1.37 of the core instant-messaging plugin.

While it successfully adds the `customMessage` support to the IRC notification (pipeline `ircNotify` at least, as tested), the solution is not quite streamlined (and also the `extraMessage` support is lacking). These will be improved in a next PR that would depend on a newer revision (eventually release) of the core instant-messaging plugin.

Merging now to freeze and publish a working improvement, before I break it again for a next iteration ;)

**Pull request merged**